modules/custom-css/custom-css/js/customizer-control.js

```diff
@@ -0,0 +1,198 @@
+(function(wp, $){
+	var api = wp.customize;
+
+	api.controlConstructor.jetpackCss = api.Control.extend({
+		modes: {
+			'default': 'text/css',
+			'less': 'text/x-less',
+			'sass': 'text/x-scss'
+		},
+		_updating: false,
+		/**
+		 * Fires when our control is ready for action. Gets everything set up.
+		 * @return {null}
+		 */
+		ready: function() {
+			this.opts = window._jp_css_settings;
+			// add our textarea
+			this.$input = $( '<textarea />', {
+				name: this.setting.id,
+				'class': 'for-codemirror hidden'
+			} ).val( this.setting() );
+			this.container.append( this.$input );
+
+			// keep the textarea and the setting synced up
+			api( this.setting.id, _.bind( function( setting ){
+				var element = new api.Element( this.$input );
+				this.elements = [ element ];
+				element.sync( setting );
+				element.set( setting() );
+			}, this ) );
+
+			// should we use CodeMirror?
+			if ( this.opts.useRichEditor ) {
+				this.initCodeMirror();
+			} else {
+				this.$input.removeClass( 'hidden' );
+			}
+
+			api.bind( 'ready', _.bind( this.addLabels, this ) );
+		},
+		/**
+		 * Set up our CodeMirror instance
+		 * @return {null}
+		 */
+		initCodeMirror: function() {
+			this.editor = window.CodeMirror.fromTextArea( this.$input.get(0), {
+				mode: this.getMode(),
+				lineNumbers: true,
+				tabSize: 2,
+				indentWithTabs: true,
+				lineWrapping: true
+			} );
+			this.setEditorHeight();
+
+			this.addListeners();
+		},
+		setEditorHeight: function() {
+			this.editor.setSize( null, $(document).height() * 0.5 );
+		},
+		/**
+		 * Adds various listeners for CodeMirror to render and keep in sync
+		 * with the textarea.
+		 */
+		addListeners: function() {
+			var edited = false;
+
+			// refresh the CodeMirror instance's rendering becuase it's initially hidden
+			// 250ms because that's the open animation duration
+			$( '#accordion-section-jetpack_custom_css > .accordion-section-title' ).click( _.bind( _.debounce( this.editor.refresh, 250 ), this.editor ) );
+			// also refresh when focusing
+			this.editor.on( 'focus', function( editor ) {
+				editor.refresh();
+			});
+
+			// React to window resizing for CodeMirror height
+			$( window ).resize( _.bind( _.debounce( this.setEditorHeight, 100 ), this ) );
+
+			// when the CodeMirror instance changes, mirror to the textarea,
+			// where we have our "true" change event handler bound. This allows both to function.
+			this.editor.on( 'change', _.bind( function( editor ) {
+				this._updating = true;
+				this.$input.val( editor.getValue() ).trigger( 'change' );
+				this._updating = false;
+
+				if ( ! edited ) {
+					window.ga && window.ga( 'send', 'event', 'Customizer', 'Typed Custom CSS' );
```

Comment:
Any chance I can impose on you for help with that?  It's a bit out of my wheelwell.